src/art.rs

```diff
@@ -19,33 +19,45 @@ const NODE48MAX: usize = 48;
 
 // Minimum and maximum number of children for Node256
 const NODE256MIN: usize = NODE48MAX + 1;
-const NODE256MAX: usize = 256;
 
 // From the specification: Adaptive Radix tries consist of two types of nodes:
 // Inner nodes, which map prefix(prefix) keys to other nodes,
 // and leaf nodes, which store the values corresponding to the key.
-struct Node<P: Prefix + Clone, V> {
+struct Node<P: Prefix + Clone, V: Clone> {
     pub node_type: NodeType<P, V>, // Type of the node
 }
 
-enum NodeType<P: Prefix + Clone, V> {
+enum NodeType<P: Prefix + Clone, V: Clone> {
     // Leaf node of the adaptive radix trie
     Leaf(LeafNode<P, V>),
     // Inner node of the adaptive radix trie
     Node4(FlatNode<P, Node<P, V>, 4>), // Node with 4 keys and 4 children
     Node16(FlatNode<P, Node<P, V>, 16>), // Node with 16 keys and 16 children
-    Node48(Node48<P, Node<P, V>, 48>), // Node with 256 keys and 48 children
+    Node48(Node48<P, Node<P, V>>),     // Node with 256 keys and 48 children
     Node256(Node256<P, Node<P, V>>),   // Node with 256 keys and 256 children
 }
 
 // Adaptive radix trie
-pub struct Tree<P: PrefixTrait, V> {
-    root: Option<Node<P, V>>, // Root node of the tree
-    size: u64,                // Number of elements in the tree
+pub struct Tree<P: PrefixTrait, V: Clone> {
+    root: Option<Arc<Node<P, V>>>, // Root node of the tree
```

Comment:
Having a Arc on root node should be enough when doing CoW, but need some clear understanding on this